protostar/profiler/transaction_profiler.py

```diff
@@ -0,0 +1,211 @@
+# pylint: disable=invalid-name
+from dataclasses import dataclass, replace
+import itertools
+from typing import TYPE_CHECKING, Tuple
+
+from protostar.profiler.contract_profiler import (
+    Instruction,
+    Function,
+    Sample,
+)
+
+
+if TYPE_CHECKING:
+    from protostar.starknet.cheatable_execute_entry_point import (
+        ContractProfile,
+        ContractFilename,
+    )
+
+Address = int
+GlobalFunctionID = str
+GlobalInstructionID = str
+
+GlobalFunction = Function
+GlobalInstruction = Instruction
```

Comment:
Can functions be local?

Reply:
Yep, local in terms contract, global in terms of transaction

Reply:
So `GlobalFunction` == `Entrypoint`?